controllers/csidriver.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022 Red Hat OpenShift Data Foundation.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	v1k8scsi "k8s.io/api/storage/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/storage/v1"
+)
+
+type csiDriver struct {
+	name           string
+	fsGroupPolicy  string
+	attachRequired bool
+	mountInfo      bool
+	csiDriver      *v1k8scsi.CSIDriver
+	csiClient      v1.CSIDriverInterface
+}
+
+// createCSIDriverInfo Registers CSI driver by creating a CSIDriver object
+func (d csiDriver) createCSIDriverInfo(ctx context.Context) error {
```

Comment:
I don't see anything to say why this needs to be its own function. I think it can be boiled down to:
```
csiDriver := getSomeCsiDriver()
if shouldDelete(csiDriver) {
  Delete(csiDriver)
}
CreateorUpdate(csiDriver)
```

In this example, `shouldDelete()` would just return the calculation of `driver.Spec.FSGroupPolicy != nil && csiDriver.Spec.FSGroupPolicy != nil && *driver.Spec.FSGroupPolicy != *csiDriver.Spec.FSGroupPolicy` from below.

At this point, as I mentioned above, if you just duplicate the code I outlined, the `getSomeCsiDriver()` functions can be really really simple and just return the relevant CsiDriver without doing a bunch of conditionals.

Reply:
rearranged and changed few things to look the same way

Reply:
:+1: